src/index/ivf/ivf_config.h

```diff
@@ -67,7 +67,8 @@ class IvfPqConfig : public IvfConfig {
     CFG_INT nbits;
     KNOHWERE_DECLARE_CONFIG(IvfPqConfig) {
         KNOWHERE_CONFIG_DECLARE_FIELD(m).description("m").for_train().set_range(1, 65536);
-        KNOWHERE_CONFIG_DECLARE_FIELD(nbits).description("nbits").set_default(8).for_train().set_range(1, 64);
+        // FAISS rejects nbits > 24, because it is not practical
+        KNOWHERE_CONFIG_DECLARE_FIELD(nbits).description("nbits").set_default(8).for_train().set_range(1, 24);
```

Comment:
1. Did you mean Faiss report error for `nbits > 24` and our E2E test has this case? Then why it didn't fail our E2E test?
2. Do we need to take care of the exist user? I prefer not.

Reply:
1. this is what we'll need to migrate from the faiss baseline https://github.com/facebookresearch/faiss/blob/1a799d014b07aea1fca6788691078054f7865f4c/faiss/impl/ProductQuantizer.cpp#L64 eventually
2. NO USER uses even 16 bits

Reply:
> 1. this is what we'll need to migrate from the faiss baseline https://github.com/facebookresearch/faiss/blob/1a799d014b07aea1fca6788691078054f7865f4c/faiss/impl/ProductQuantizer.cpp#L64 eventually
> 2. NO USER uses even 16 bits

Then, looks like you need to modify `knowhere-test` or all other PR will suffer